velox/functions/sparksql/Register.cpp

```diff
@@ -146,6 +146,74 @@ void registerFunctions(const std::string& prefix) {
+      {prefix + "millisecond"});
+  registerFunction<MillisecondFunction, int32_t, Timestamp>(
+      {prefix + "millisecond"});
+  registerFunction<MillisecondFunction, int32_t, TimestampWithTimezone>(
+      {prefix + "millisecond"});
+  registerFunction<SecondFunction, int32_t, Date>(
+      {prefix + "second"});
+  registerFunction<SecondFunction, int32_t, Timestamp>(
+      {prefix + "second"});
+  registerFunction<SecondFunction, int32_t, TimestampWithTimezone>(
+      {prefix + "second"});
+  registerFunction<MinuteFunction, int32_t, Date>(
+      {prefix + "minute"});
+  registerFunction<MinuteFunction, int32_t, Timestamp>(
+      {prefix + "minute"});
+  registerFunction<MinuteFunction, int32_t, TimestampWithTimezone>(
+      {prefix + "minute"});
+  registerFunction<HourFunction, int32_t, Date>(
+      {prefix + "hour"});
+  registerFunction<HourFunction, int32_t, Timestamp>(
+      {prefix + "hour"});
+  registerFunction<HourFunction, int32_t, TimestampWithTimezone>(
+      {prefix + "hour"});
+  registerFunction<DayFunction, int32_t, Date>(
+      {prefix + "day", prefix + "day_of_month"});
+  registerFunction<DayFunction, int32_t, Timestamp>(
+      {prefix + "day", prefix + "day_of_month"});
+  registerFunction<DayFunction, int32_t, TimestampWithTimezone>(
+      {prefix + "day", prefix + "day_of_month"});
+  registerFunction<DayOfWeekFunction, int32_t, Date>(
+      {prefix + "day_of_week"});
+  registerFunction<DayOfWeekFunction, int32_t, Timestamp>(
+      {prefix + "day_of_week"});
+  registerFunction<DayOfWeekFunction, int32_t, TimestampWithTimezone>(
+      {prefix + "day_of_week"});
+  registerFunction<DayOfYearFunction, int32_t, Date>(
+      {prefix + "day_of_year"});
+  registerFunction<DayOfYearFunction, int32_t, Timestamp>(
+      {prefix + "day_of_year"});
+  registerFunction<DayOfYearFunction, int32_t, TimestampWithTimezone>(
+      {prefix + "day_of_year"});
+  registerFunction<MonthFunction, int32_t, Date>(
+      {prefix + "month"});
+  registerFunction<MonthFunction, int32_t, Timestamp>(
+      {prefix + "month"});
+  registerFunction<MonthFunction, int32_t, TimestampWithTimezone>(
+      {prefix + "month"});
+  registerFunction<QuarterFunction, int32_t, Date>(
+      {prefix + "quarter"});
+  registerFunction<QuarterFunction, int32_t, Timestamp>(
+      {prefix + "quarter"});
+  registerFunction<QuarterFunction, int32_t, TimestampWithTimezone>(
+      {prefix + "quarter"});
+  registerFunction<YearFunction, int32_t, Date>(
+      {prefix + "year"});
+  registerFunction<YearFunction, int32_t, Timestamp>(
+      {prefix + "year"});
+  registerFunction<YearFunction, int32_t, TimestampWithTimezone>(
+      {prefix + "year"});
+  registerFunction<YearOfWeekFunction, int32_t, Date>(
+      {prefix + "year_of_week"});
+  registerFunction<YearOfWeekFunction, int32_t, Timestamp>(
+      {prefix + "year_of_week"});
+  registerFunction<YearOfWeekFunction, int32_t, TimestampWithTimezone>(
+      {prefix + "year_of_week"});
```

Comment:
Are you going to open PR to Velox to add these functions to spark sql?

Reply:
Yes, I've opened this PR(https://github.com/facebookincubator/velox/pull/3362)